addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -89,38 +90,34 @@
         public void run() {
             try {
                 pollingLock.lock();
-                try {
-                    if (!lastBridgeConnectionState) {
-                        // if user is not set in configuration try to create a new user on Hue bridge
-                        if (hueBridgeConfig.getUserName() == null) {
-                            hueBridge.getFullConfig();
-                        }
-                        lastBridgeConnectionState = tryResumeBridgeConnection();
-                    }
-                    if (lastBridgeConnectionState) {
-                        doConnectedRun();
+                if (!lastBridgeConnectionState) {
+                    // if user is not set in configuration try to create a new user on Hue bridge
+                    if (hueBridgeConfig.getUserName() == null) {
+                        hueBridge.getFullConfig();
                     }
-                } catch (UnauthorizedException | IllegalStateException e) {
-                    if (isReachable(hueBridge.getIPAddress())) {
+                    lastBridgeConnectionState = tryResumeBridgeConnection();
+                }
+                if (lastBridgeConnectionState) {
+                    doConnectedRun();
+                }
+            } catch (UnauthorizedException | IllegalStateException e) {
+                if (isReachable(hueBridge.getIPAddress())) {
+                    lastBridgeConnectionState = false;
+                    onNotAuthenticated();
+                } else {
+                    if (lastBridgeConnectionState || thing.getStatus() == ThingStatus.INITIALIZING) {
                         lastBridgeConnectionState = false;
-                        onNotAuthenticated();
-                    } else {
-                        if (lastBridgeConnectionState || thing.getStatus() == ThingStatus.INITIALIZING) {
-                            lastBridgeConnectionState = false;
-                            onConnectionLost();
-                        }
+                        onConnectionLost();
                     }
-                } catch (Exception e) {
-                    if (hueBridge != null) {
-                        if (lastBridgeConnectionState) {
-                            logger.debug("Connection to Hue Bridge {} lost.", hueBridge.getIPAddress());
-                            lastBridgeConnectionState = false;
-                            onConnectionLost();
-                        }
+                }
+            } catch (ApiException | IOException e) {
+                if (hueBridge != null) {
+                    if (lastBridgeConnectionState) {
+                        logger.debug("Connection to Hue Bridge {} lost.", hueBridge.getIPAddress());
+                        lastBridgeConnectionState = false;
+                        onConnectionLost();
                     }
                 }
-            } catch (Throwable t) {
```

Comment:
Since this is in a scheduled thread you still might want to catch `RuntimException` because it will stop the scheduled job. Logging only is not needed as that would already be done by the scheduler, but it might mean the status of the thing needs to be set to offline so the user knows something is wrong. Otherwise it will silently stop and no updates will be done anymore.

Reply:
Any comments on my comment about poll thread being killed on `RuntimException`?

Reply:
Yes, I added a check for a `RuntimeException` in [line 111](https://github.com/openhab/openhab2-addons/pull/4805/files#diff-aed2b9a027c7a288f2ba9684da66b8d0R111). Do you think that is sufficient?

Reply:
Ah, I missed that. It will do the job, but when the `RuntimeException` is thrown it is possible a bug or maybe something external has changed. Thus the message displayed to the user is not completely clear now. Also you should log the exception to the log (maybe warn) because it's something unexpected so it would be really helpful to find out what is triggering the exception. So I think it would be better handled in a different catch block.

Reply:
Makes sense.